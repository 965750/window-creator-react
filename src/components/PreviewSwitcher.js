import React, { Component } from 'react'
import styles from './PreviewSwitcher.module.scss'

export default function PreviewSwitcher({ onSwitchPreview, interiorPreview }) {
    return (
        <div className='flex cursor-pointer text-center absolute top-10 right-20'>
            <div onClick={() => onSwitchPreview(true)} className={`${styles.sideBtn} ${interiorPreview ? 'border-2' : 'border bg-botticelli opacity-75'} border-gullGray`}>
                3D
            </div>
            <div onClick={() => onSwitchPreview(false)} className={`${styles.sideBtn} ${!interiorPreview ? 'border-2' : 'border bg-botticelli opacity-75'} border-gullGray`}>
                2D
            </div>
        </div>
    )
}
