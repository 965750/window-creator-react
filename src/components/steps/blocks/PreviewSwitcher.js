import React, { Component } from 'react'
import styles from './PreviewSwitcher.module.scss'
import PropTypes from 'prop-types'

function PreviewSwitcher({ onSwitchPreview, interiorPreview }) {
    return (
        <div className="flex cursor-pointer text-center absolute top-10 right-20">
            <div
                onClick={() => onSwitchPreview(true)}
                className={`${styles.sideBtn} ${
                    interiorPreview
                        ? 'border-2 bg-white'
                        : 'border bg-botticelli opacity-75'
                } border-gullGray`}
            >
                3D
            </div>
            <div
                onClick={() => onSwitchPreview(false)}
                className={`${styles.sideBtn} ${
                    !interiorPreview
                        ? 'border-2 bg-white'
                        : 'border bg-botticelli opacity-75'
                } border-gullGray`}
            >
                2D
            </div>
        </div>
    )
}

PreviewSwitcher.propTypes = {
    onSwitchPreview: PropTypes.func,
    interiorPreview: PropTypes.bool,
}

export default PreviewSwitcher
