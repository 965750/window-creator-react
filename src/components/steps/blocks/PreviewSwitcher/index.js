import PropTypes from 'prop-types'
import React from 'react'
import styles from './PreviewSwitcher.module.scss'

function PreviewSwitcher({ onSwitchPreview, interiorPreview }) {
  return (
    <div className="flex cursor-pointer text-center absolute top-10 right-20">
      <button
        data-testid="previewSwitcher3d"
        onClick={() => onSwitchPreview(true)}
        className={`${styles.sideBtn} ${
          interiorPreview
            ? 'border-2 bg-white'
            : 'border bg-botticelli opacity-75'
        } border-gullGray`}
      >
        3D
      </button>
      <button
        data-testid="previewSwitcher2d"
        onClick={() => onSwitchPreview(false)}
        className={`${styles.sideBtn} ${
          !interiorPreview
            ? 'border-2 bg-white'
            : 'border bg-botticelli opacity-75'
        } border-gullGray`}
      >
        2D
      </button>
    </div>
  )
}

PreviewSwitcher.propTypes = {
  onSwitchPreview: PropTypes.func,
  interiorPreview: PropTypes.bool,
}

export default PreviewSwitcher
