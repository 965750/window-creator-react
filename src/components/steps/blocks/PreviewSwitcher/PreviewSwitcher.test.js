import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import PreviewSwitcher from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('PreviewSwitcher', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <PreviewSwitcher {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should show active 3D button when interiorPreview is active', () => {
    setUp({
      interiorPreview: true
    })

    expect(wrapper.queryByTestId('previewSwitcher3d')).toHaveClass('border-2 bg-white')
  })

  it('should show active 2D button when interiorPreview is NOT active', () => {
    setUp({
      interiorPreview: false
    })

    expect(wrapper.queryByTestId('previewSwitcher2d')).toHaveClass('border-2 bg-white')
  })
})
