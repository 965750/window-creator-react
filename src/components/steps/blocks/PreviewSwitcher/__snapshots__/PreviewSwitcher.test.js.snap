// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewSwitcher should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex cursor-pointer text-center absolute top-10 right-20"
  >
    <button
      class="sideBtn border bg-botticelli opacity-75 border-gullGray"
      data-testid="previewSwitcher3d"
    >
      3D
    </button>
    <button
      class="sideBtn border-2 bg-white border-gullGray"
      data-testid="previewSwitcher2d"
    >
      2D
    </button>
  </div>
</DocumentFragment>
`;
